marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -142,7 +146,20 @@ protected JsonNode determineIngestionEndpointParams(Map<String, String> options)
         }
 
         if (doesNotHaveApiPath) {
-            endpointParams.put("apiPath", "/data-hub/5/data-services/ingestion/bulkIngester.api");
+            String apiPath = "/data-hub/5/data-services/ingestion/bulkIngester.api";
+
+            if(hubClient.getModulesClient().newJSONDocumentManager().exists(apiPath) == null) {
+                IOUtil ioUtil = new IOUtil(this.hubClient.getModulesClient());
+                try {
+                    ioUtil.load("bulkIngester.api");
+                } catch (IOException e) {
+                    throw new RuntimeException("Error occurred while loading default endpoints.");
```

Comment:
We don't want to lose the original exception here - e.g:

    throw new RuntimeException("Unable to write default ingestion endpoint at path: " + path + "; cause: " + e.getMessage(), e);